flytekit/types/directory/types.py

```diff
@@ -512,8 +518,9 @@ def to_python_value(
 
         # This is a local file path, like /usr/local/my_dir, don't mess with it. Certainly, downloading it doesn't
         # make any sense.
+        # Turn to true for pydantic plugin
         if not ctx.file_access.is_remote(uri):
-            return expected_python_type(uri, remote_directory=False)
+            return expected_python_type(uri)
```

Comment:
We have to do this to upload dir to remote storage.

Reply:
When `pydantic` create `FlyteDirectory`, it will call `to_python_value` to create a python instance when the first time we initialize FlyteDirectory, which will set `remote_directory ` variable to false in `FlyteDirectory`.
When we called `to_literal`, since `remote_directory=False`, `upload` variable will be false, and it will not be uploaded.

Reply:
There's 1 more alternative to achieve this:
Add a parameter in user space, but I think it will make the code more messy.